core-skills/frontend-web-development/resources/css-architecture.md

```diff
@@ -1,23 +1,67 @@
-- https://developer.mozilla.org/en-US/docs/Learn/CSS/CSS_layout
-- https://developer.mozilla.org/en-US/docs/Learn/CSS/CSS_layout/Floats 
-- https://developer.mozilla.org/en-US/docs/Learn/CSS/CSS_layout/Flexbox 
-- https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Flexible_Box_Layout/Basic_Concepts_of_Flexbox 
-- https://hacks.mozilla.org/2016/04/you-might-not-need-a-css-framework/
-- https://css-tricks.com/what-are-the-benefits-of-using-a-css-framework/
+# CSS Architecture
 
-## Semantic CSS
+A well-written set of stylesheets balance a number of factors including
+
+* Browser support - does the styling render the same in a wide range of commonly used browsers?
+* Ease of change - can we easily change the styles to meet changing customer needs?
+* Reuse - can we reuse commonly used styles easily?
+* Stability - can we safely change our styles without worrying about unrelated breakages?
+* Expressiveness - can we understand what our styles do, and do they make sense?
+* Performance - are our styles fast to load and render?
+
+If this seems overwhelming, don't worry, we're here to help.
+
+## CSS Resets
+
+One issue you will face is that even with a brand-new HTML file, without any stylesheets imported, different browsers will render this file differently. The cause of this is differing browser-default styling.
+
+The defacto-accepted solution to this is to use a CSS reset stylesheet (of which there are many). A CSS reset will ensure that all browsers have the same basic styles upon which you can start layering custom styles ontop of.
+
+**Note:** most CSS frameworks prepackage a CSS reset.
+
+## CSS Methodologies
+
+You will also need to choose an approach to achieve a well-architected set of CSS styles.
+
+### Component Based CSS
+
+If you are using React (or similar), it is likely that a Component-based approach makes the most sense
+
+### BEM
+
+Block Element Modifier.
 
-- https://maintainablecss.com/chapters/semantics/
-- https://css-tricks.com/when-using-important-is-the-right-choice/
-- https://css-tricks.com/snippets/css/a-guide-to-flexbox/
 - http://getbem.com/
+
+### we also give a mention to...
```

Comment:
Wondering whether we should be a little bit opinionated about BEM and Component based (for beginners). Just so people aren't given a paradox of choice